src/tools/pub/cmd/pub-outdated.md

```diff
@@ -0,0 +1,251 @@
+
+```diff
+-  build_web_compilers: ^1.0.0
++  build_web_compilers: ^2.10.0
+```
+
+{{site.alert.tip}}
+  To see what changed in the new version, you can look at the package's changelog,
+  which you can find in the package page on
+  [pub.dev.]({{site.pub}})
+  For an example, see the
+  [build_web_compilers changelog.]({{site.pub-pkg}}/build_web_compilers#-changelog-tab-)
+{{site.alert.end}}
+
+After editing `pubspec.yaml`, run `pub upgrade` to
+update the `pubspec.lock` file.
+You can then run `pub outdated` to confirm that
+you've made all necessary changes.
+
+```terminal
+$ pub upgrade
+...
+$ pub outdated
+Package     Current  Upgradable  Resolvable  Latest  
+dependencies: all up-to-date
+
+dev_dependencies: all up-to-date
+
+transitive dependencies
+googleapis  0.53.0   0.53.0      0.53.0      0.54.0  
+$
+```
+<span style="color:red">[PENDING: file bug about "up-to-date" (shouldn't be hyphenated unless it comes before a noun)]
+
+{{site.alert.important}}
+  Test your code to verify that it still works as expected
+  after updating the packages.
+{{site.alert.end}}
+
+If any transitive dependencies are out of date,
+the best way to update those for package maintainers
+to update their code.
+You can find out the cause of the out-of-date dependency
+by running [`pub deps`][]
+and searching the output for the name of the out-of-date package.
+Then make sure an issue exists for the out-of-date dependency;
+you can find the issue tracker for a package
+by looking at its pub page.
+<span style="color:red">
+  [PENDING: What exactly do we want to recommend?
+  Searching for `googleapis` in the `pub deps` output tells me that
+  `sdk_builds 0.2.0-dev` depends on `googleapis 0.53.0.`
+  But although https://pub.dev/packages/sdk_builds exists,
+  I think it's just referring to the Dart SDK.
+  So we can't say everything's a package, can we?]
+</span>
+
+[`pub deps`]: /tools/pub/cmd/pub-deps)
+[`pub get`]: /tools/pub/cmd/pub-get)
+[`pub upgrade`]: /tools/pub/cmd/pub-upgrade)
+
+
+## Using the output
+
+As a reminder, here's what the output of `pub outdated` looks like:
+
+
+```terminal
+$ pub outdated
+Package              Current   Upgradable  Resolvable  Latest    
+dependencies: all up-to-date
+
+dev_dependencies    
+build_web_compilers  1.2.3     1.2.3       2.10.0      2.10.0    
+
+transitive dependencies
+_fe_analyzer_shared  -         -           2.0.0       2.0.0     
+analyzer             0.35.4    0.35.4      0.39.5      0.39.5    
+...
+
+9  dependencies are constrained to versions that are older than a resolvable version.
+To update these dependencies, edit pubspec.yaml.
+```
+
+As a rule, update the **dependencies** and **dev_dependencies** sections
+of your `pubspec.yaml` file
+to depend on the **resolvable** versions of out-of-date packages.
+<span style="color:red">
+  [PENDING: correct? explain why?]
+</span>
+
+Here are the meanings of the version numbers in the output:
```

Comment:
Oh, here is what I had in mind. I think I would move this up before the first example.

Reply:
I think it's important to have the example up high in the page, but I've added a link to this section.